### ISSUE TITLE
React scale-app-header can't open second level nav with keyboard

### ISSUE THREAD

**Issue by christopherGdynia** [telekom/scale#384]
Hello there,

in React it isnt possible to open the second level navigation in main navigation.
It should be a react problem, in your storybook it is working.

Best regards 
Christopher

Code: 

jsx```

import { ScaleAppShell, ScaleAppFooter } from "@telekom/scale-components-react";
import {} from "@telekom/scale-components/hydrate";

import React from "react";

interface IHeaderProps {
  activeRouteId?: string;
}

const Header: React.FC<IHeaderProps> = ({ activeRouteId, children }) => {
  const iconNavigation = [
    { name: "Search", id: "Search41", href: "#search", icon: "action-search" },
    {
      name: "Profil",
      id: "Profil42",
      href: "#profile",
      icon: "user-file-user",
    },
    {
      name: "Ausloggen",
      id: "Logout43",
      href: "#logout",
      icon: "user-file-logout",
    },
    { id: "menu", defaultName: "Menu", openedName: "Close" },
  ];

  const mainNavigation = [
    { name: "Startseite", id: "startseite", href: "/" },
    {
      name: "Apps",
      id: "apps",
      children: [
        {
          name: "Voice",
          id: "voice",
          children: [
            {
              name: "ABM",
              id: "abm",
              href: "https://abm.telekom.de/serv",
              target: "_blank",
            },
          ],
        },
        {
          name: "ÜWege",
          id: "uwege",
          children: [
            {
              name: "ÜWege Factory",
              id: "factory",
              href: "https://factory.telekom.de",
              target: "_blank",
            },
          ],
        },
        {
          name: "Übergreifend",
          id: "ubergreifend",
          children: [
            {
              name: "COOL",
              id: "cool",
              href: "https://cool.telekom.de",
              target: "_blank",
            },
          ],
        },
      ],
    },
    { name: "Impressum", id: "impressum", href: "/impressum" },
  ];
  return (
    <>
      <ScaleAppShell
        activeRouteId={activeRouteId}
        logoTitle="Zur Startseite"
        logoHref="/"
        mainNavigation={mainNavigation}
        iconNavigation={iconNavigation}
        portalName={"Listen Organisation Tools Und Services"}
        claimLang={"de"}
      >
        {children}
      </ScaleAppShell>
      <ScaleAppFooter
        footerNavigation={[
          { name: "Startseite", id: "startseite", href: "/" },
          { name: "Impressum", id: "impressum", href: "/impressum" },
          { name: "Datenschutz", id: "datenschutz", href: "/datenschutz" },
        ]}
        claimLang={"de"}
        copyright={"© Prozess- & Portfoliotransformation SGrK"}
      />
    </>
  );
};
```

EDIT: 

Pressing space on the focused element in Navigation should open the second level navigation, but its scrolling down for some reason

**Comment by acstll** [comment]
Thank you @christopherGdynia 

We can try and figure out whether this is actually a React problem or something we can fix.

Hopefully @nowseemee will be able to take a look next week.

---

Regarding the space bar and scrolling, it's a browser convention that allows navigation without a mouse. If there's no other function assign to it, pressing Space should scroll down, and pressing Shift + Space should scroll up.

**Comment by nowseemee** [comment]
Hi @christopherGdynia,

right on, another bug in the header - many thanks for opening this issue and making me aware.

I've just patched the bug and released `3.0.0-beta.13`.

A little heads-up: `Enter` and `Esc` keys will toggle the highlighted main menu entry's secondary navigation - `Spacebar` is not mapped for this use-case.

Cheers,
Daniel
